src/routes/api/users.js

```diff
@@ -1,82 +1,25 @@
 import { Router } from 'express';
 import passport from 'passport';
+import userController from '../../controllers/userController';
 
 const router = Router();
 
-router.get('/user', (req, res, next) => {
-  User.findById(req.payload.id)
-    .then((user) => {
-      if (!user) {
-        return res.sendStatus(401);
-      }
-      return res.json({ user: user.toAuthJSON() });
-    })
-    .catch(next);
-});
+const { forgotPassword, resetPassword } = userController;
 
-router.put('/user', (req, res, next) => {
-  User.findById(req.payload.id)
-    .then((user) => {
-      if (!user) {
-        return res.sendStatus(401);
-      }
+router.get('/user');
 
-      // only update fields that were actually passed...
-      if (typeof req.body.user.username !== 'undefined') {
-        user.username = req.body.user.username;
-      }
-      if (typeof req.body.user.email !== 'undefined') {
-        user.email = req.body.user.email;
-      }
-      if (typeof req.body.user.bio !== 'undefined') {
-        user.bio = req.body.user.bio;
-      }
-      if (typeof req.body.user.image !== 'undefined') {
-        user.image = req.body.user.image;
-      }
-      if (typeof req.body.user.password !== 'undefined') {
-        user.setPassword(req.body.user.password);
-      }
+router.put('/user');
 
-      return user.save().then(() => res.json({ user: user.toAuthJSON() }));
-    })
-    .catch(next);
-});
+router.post('/users');
 
-router.post('/users/login', (req, res, next) => {
-  if (!req.body.user.email) {
-    return res.status(422).json({ errors: { email: "can't be blank" } });
-  }
+// @route POST /api/v1/users/forgotpassword
+// @desc Generate User Password Reset / Returning JWT Token
+// @access Public
+router.post('/forgotpassword', forgotPassword);
```

Comment:
your route naming doesn't follow the convention I discussed yesterday. you should have something like this `/users/passwords/forgot` and `/users/passwords/reset`.

Reply:
routes reviewed as advised.